pkg/reconciler/taskrun/local.go

```diff
@@ -0,0 +1,57 @@
+package taskrun
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Local struct{}
+
+func (l Local) Allocate(r *ReconcileTaskRun, ctx context.Context, tr *pipelinev1.TaskRun, secretName string) (reconcile.Result, error) {
+	var err error
+	log := logr.FromContextOrDiscard(ctx)
+
+	log.Info("Task set to run locally in the cluster")
+	tr.Labels[AssignedHost] = "localhost"
+	controllerutil.AddFinalizer(tr, PipelineFinalizer)
+	if err = r.client.Update(ctx, tr); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = createUserTaskSecret(r, ctx, tr, secretName, map[string][]byte{
+		"host": []byte("localhost"),
+	})
+	return reconcile.Result{}, err
+}
+
+func (l Local) Deallocate(r *ReconcileTaskRun, ctx context.Context, tr *pipelinev1.TaskRun, secretName string, selectedHost string) error {
+	return nil
+}
+
+func createUserTaskSecret(r *ReconcileTaskRun, ctx context.Context, tr *pipelinev1.TaskRun, secretName string, secretData map[string][]byte) error {
```

Comment:
Is this the function that is needed for the OTP functionality?

Reply:
No, there is no OTP involved for the "local" hosts. This is for creating the secret the user task will mount. For actual remote hosts the secret is indeed generated via the OTP service, but that is done with a shell script running in a Task elsewhere.